src/stim/mem/simd_bits.test.cc

```diff
@@ -228,6 +228,22 @@ TEST_EACH_WORD_SIZE_W(simd_bits, add_assignment, {
     m0 += m1;
     ASSERT_EQ(m0[0], 0);
     ASSERT_EQ(m0[64], 1);
+    // Test carrying across multiple (>=2) words.
+    size_t num_bits = 193;
+    simd_bits<W> add(num_bits);
+    simd_bits<W> one(num_bits);
+    for (size_t word = 0; word < add.num_u64_padded() - 1; word++) {
+        for (size_t k = 0; k < 64; k++) {
+            add[word * 64 + k] = 1;
+        }
+    }
+    one[0] = 1;
+    add += one;
+    // These should all overflow and carries should propagate to the last word.
+    for (size_t k = 0; k < num_bits - 1; k++) {
+        ASSERT_EQ(add[k], 0);
+    }
```

Comment:
Given that this failed, please add a fuzz test. Bias the fuzz test towards interesting cases by with uniform 1/3 chance for each word making it all-1 or all-0 or all-random.

Reply:
Is testing by adding bit-by-bit using [] sufficient here, or is testing against python preferred, second way requires exposing to pybind?

Reply:
Ah, fair point, you need a ground truth to compare against, and it has to come from code that's unlikely to have the same bugs.... hmmm....

Instead of checking the exact answer, you could check properties like commutativity and associativity. You could verify that `a += x; a = ~a; a += x; a = ~a` results in `a` being unchanged. You could check that `a + a == a << 1` and that `a + ~a == 1`. You could check that `a + (b & c) + (b ^ c) == a + b + c`. Anything that sortof rearranges the problem in a way that would trigger the boundary issues differently on the left hand side and right hand side.

And yeah, pulling in a few hardcoded cases from python would also work. I wouldn't expose simd_bits all the way up to python.

Reply:
Cool, good suggestions!